rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -256,16 +257,21 @@ impl ExecutionPlan for ParquetExec {
         let projection = self.projection.clone();
         let batch_size = self.batch_size;
 
-        thread::spawn(move || {
-            if let Err(e) = read_files(
+        task::spawn_blocking(move || {
+            read_files(
                 &filenames,
                 projection.clone(),
                 batch_size,
                 response_tx.clone(),
-            ) {
-                println!("Parquet reader thread terminated due to error: {:?}", e);
-            }
-        });
+            )
+        })
+        .await
```

Comment:
I don't think we want to `await` here for the task to complete?

Reply:
I think this will lead to deadlock because the channel's buffer will fill up and the reader won't be able to push new batches. The caller can't read any batches from the channel until this task completes?